src/array/cuda/dispatcher.cuh

```diff
@@ -0,0 +1,131 @@
+/*!
+ *  Copyright (c) 2020 by Contributors
+ * \file array/cuda/dispatcher.cuh
+ * \brief Dispatches to CUDA functions by data types.
```

Comment:
Is this header for compatibility of different cusparse versions? If so, please say it explicitly (I would recommend change the file name as well). Right now, it looks like a header file for general dispatchers.

Reply:
No it's a dispatcher for different input types.  It dispatches float inputs to cusparseScsrgemm2 and double inputs to cusparseDcsrgemm2 for instance.

Reply:
I understand. I would name it `cusparse_dispatcher.h` and put down description like "Templates to call different CUSPARSE routines based on the type argument" or similar sorts.

Reply:
Done.